From 29db2843aa89c9eb5966ec74a5d1e376f37b357c Mon Sep 17 00:00:00 2001
From: Gunaal R <quic_gunar@quicinc.com>
Date: Tue, 27 Aug 2024 16:03:19 +0530
Subject: [PATCH] QSDK: ath12k: add WARN ON for wait peer delete.

Once peer delete is sent from host to firmware, host will wait
for peer unmap event and peer delete response from firmware.
When the events are not received, though host cleans up the
internal  peer list, it will send failure status to mac80211
which has WARN_ON on failure case:

To debug such scenario from firmware, WARN_ON is added when host
timed out waiting for the events. This will help in generating
required dumps for further debugging from firmware team.

[ 1494.125212] Call trace:
[ 1494.132325]  sta_set_sinfo+0xa28/0xab0 [mac80211]
[ 1494.134583]  __sta_info_destroy+0x2c/0x3c [mac80211]
[ 1494.139445]  sta_info_destroy_addr_bss+0x30/0x50 [mac80211]
[ 1494.144482]  ieee80211_nan_func_match+0x2c8c/0x5300 [mac80211]
[ 1494.149775]  cfg80211_check_station_change+0x12b4/0x1b68 [cfg80211]
[ 1494.155678]  genl_family_rcv_msg_doit+0xd0/0x13c
[ 1494.161840]  genl_rcv_msg+0x1d0/0x204
[ 1494.166699]  netlink_rcv_skb+0x9c/0x108
[ 1494.170258]  genl_rcv+0x34/0x48
[ 1494.173902]  netlink_unicast+0x168/0x244
[ 1494.177029]  netlink_sendmsg+0x35c/0x3b8
[ 1494.181197]  sock_sendmsg+0x18/0x2c
[ 1494.185102]  ____sys_sendmsg+0x2c4/0x364
[ 1494.188315]  ___sys_sendmsg+0x74/0xbc
[ 1494.192481]  __sys_sendmsg+0x64/0xb0
[ 1494.196040]  __arm64_sys_sendmsg+0x1c/0x24
[ 1494.199686]  el0_svc_common.constprop.0+0xc0/0x130
[ 1494.203592]  do_el0_svc+0x18/0x20
[ 1494.208366]  el0_svc+0x14/0x3c
[ 1494.211751]  el0t_64_sync_handler+0xa8/0x134
[ 1494.214703]  el0t_64_sync+0x14c/0x150

Patch-dependency: None
Patch-work: None

Signed-off-by: Gunaal R <quic_gunar@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/peer.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/net/wireless/ath/ath12k/peer.c b/drivers/net/wireless/ath/ath12k/peer.c
index 7506540..17dfccd 100644
--- a/drivers/net/wireless/ath/ath12k/peer.c
+++ b/drivers/net/wireless/ath/ath12k/peer.c
@@ -527,6 +527,7 @@ int ath12k_wait_for_peer_delete_done(struct ath12k *ar, u32 vdev_id,
 	ret = ath12k_wait_for_peer_deleted(ar, vdev_id, addr);
 	if (ret) {
 		ath12k_warn(ar->ab, "failed wait for peer deleted peer_addr : %pM\n", addr);
+		WARN_ON(1);
 		return ret;
 	}
 
-- 
2.34.1

